pwndbg/color/lexer.py

```diff
@@ -0,0 +1,156 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import re
+
+import six
+from pygments.lexer import RegexLexer
+from pygments.lexer import include
+from pygments.token import Comment
+from pygments.token import Name
+from pygments.token import Number
+from pygments.token import Operator
+from pygments.token import Other
+from pygments.token import Punctuation
+from pygments.token import String
+from pygments.token import Text
+
+__all__ = ['PwntoolsLexer']
+
+# Text        Token.Text            for any type of text data
+# Whitespace  Token.Text.Whitespace for specially highlighted whitespace
+# Error       Token.Error           represents lexer errors
+# Other       Token.Other           special token for data not matched by a parser (e.g. HTML markup in PHP code)
+# Keyword     Token.Keyword         any kind of keywords
+# Name        Token.Name            variable/function names
+# Literal     Token.Literal         Any literals
+# String      Token.Literal.String  string literals
+# Number      Token.Literal.Number  number literals
+# Operator    Token.Operator        operators (+, not...)
+# Punctuation Token.Punctuation     punctuation ([, (...)
+# Comment     Token.Comment         any kind of comments
+# Generic     Token.Generic         generic tokens (have a look at the explanation below)
+class PwntoolsLexer(RegexLexer):
```

Comment:
Btw have you checked how it works with other archs - e.g. MIPS?

Reply:
No, just check with ARM and a few x86-64 code with Intel syntax (NASM).

The pwnlib's lexer is modified from AT&T parser of `pygments` and add a few rule to make some environment works. It miss a lots feature for each arch. However, if we want to support multiple arch, I need someone who know each kind of asm syntax and make the new feature will not break the old one. Moreover, some syntaxes may not compatible.

I only fix some ARM feature I used.

To develop the lexer to support is for each arch, we can maintain a different lexer for the baseline. Then, we create a test for `PwntoolsLexer` force it to produce the same for different code. However, the difficulty of maintaining `PwntoolsLexer` is still the same.